File: solar_eye_backend/app/schemas/monitoring.py
from typing import List, Optional
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.schemas.detection import DetectionResponse

class AnalysisSessionBase(BaseModel):
    panel_id: int
    type: str  # "cctv" or "drone"
    status: str  # "processing", "completed", "failed"
    original_image_url: Optional[str] = None

class AnalysisSessionCreate(AnalysisSessionBase):
    pass

class AnalysisSessionResponse(AnalysisSessionBase):
    id: UUID
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class CropImageSchema(BaseModel):
    """Crop된 패널 이미지 정보"""
    url: str  # Enhanced 이미지 URL
    mask_url: Optional[str] = None  # Mask 이미지 URL
    bbox: dict  # {x, y, width, height}
    defect_type: str
    confidence: float

class AnalysisResultResponse(AnalysisSessionResponse):
    detections: List[DetectionResponse] = []
    
    # CCTV 분석 전용 필드 (lssunflower 프론트엔드 호환용)
    enhanced_image_url: Optional[str] = None  # Real-ESRGAN 결과
    mask_image_url: Optional[str] = None  # SegFormer 마스크 컬러 이미지
    crop_images: List[CropImageSchema] = []  # Crop된 패널 이미지 목록
    
    model_config = ConfigDict(from_attributes=True)

class AnalysisHistoryResponse(BaseModel):
    sessions: List[AnalysisSessionResponse]

File: solar_eye_frontend/lib/domain/model/monitoring.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:solar_eye_frontend/domain/model/detection.dart';

part 'monitoring.freezed.dart';
part 'monitoring.g.dart';

/// ID 변환 헬퍼 (int -> String)
String _idToString(dynamic value) => value.toString();

@freezed
class AnalysisSession with _$AnalysisSession {
  const factory AnalysisSession({
    required String id,
    @JsonKey(name: 'panel_id', fromJson: _idToString) required String panelId,
    required String type, // "cctv" or "drone"
    required String status, // "processing", "completed", "failed"
    @JsonKey(name: 'original_image_url') String? originalImageUrl,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @Default([]) List<Detection> detections,
  }) = _AnalysisSession;

  factory AnalysisSession.fromJson(Map<String, dynamic> json) =>
      _$AnalysisSessionFromJson(json);
}

@freezed
class AnalysisHistoryResponse with _$AnalysisHistoryResponse {
  const factory AnalysisHistoryResponse({
    required List<AnalysisSession> sessions,
  }) = _AnalysisHistoryResponse;

  factory AnalysisHistoryResponse.fromJson(Map<String, dynamic> json) =>
      _$AnalysisHistoryResponseFromJson(json);
}
